x-pack/plugins/enterprise_search/public/applications/app_search/utils/format_api_name.ts

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export function formatApiName(rawName: string) {
+  return rawName
+    .trim()
+    .replace(/[^a-zA-Z0-9]+/g, '-')
+    .replace(/^[^a-zA-Z0-9]+/, '')
+    .replace(/[^a-zA-Z0-9]+$/, '')
+    .toLowerCase();
+}
```

Comment:
Final suggestion (including preferring arrow syntax over function() for consistency):

```suggestion
export const formatApiName = (rawName: string) =>
  rawName
    .trim()
    .replace(/[^a-zA-Z0-9]+/g, '-') // Replace all special/non-alphanumerical characters with dashes
    .replace(/^[-]+|[-]+$/g, '') // Strip all leading and trailing dashes
    .toLowerCase();
};
```

Reply:
c715de8a4dbf6bbc62f7b42eb3da489a52806908